pure3.md

```diff
@@ -0,0 +1,282 @@
+a subset of ECUs may lack the resources to perform all of Scudo's verification
+workflows. Note that in the stop-gap option, _all_ ECUs lacked the necessary
+resources. Uptane already accounts for such secondary ECUs by providing a
+separate Partial Verification workflow. Scudo assumes that these Partial
+Verification ECUs are also incapable of performing in-toto verification.
+In addition, there may be Uptane Full Verification ECUs that also lack the
+resources to perform the added in-toto verification.
+
+For these ECUs, Scudo specifies that other capable ECUs, perhaps the Primary ECU,
+must perform in-toto verification on behalf of the resource constrained ECUs.
+In this model, while every ECU may not be verifying the software supply chain of
+the images it is expected to install, verification is still happening on the
+individual automobile. Essentially, there is no difference in the update
+workflow of resource constrained ECUs but they are still protected to a degree
+by the supply chain verification happening on the automobile.
+
+| Type of ECU | Uptane Responsibility | in-toto Responsibility |
+| ----------- | --------------------- | ---------------------- |
+| Primary ECU (Full Verification) | Full Uptane verification for its images and all secondaries | in-toto verification of its images as well as images of resource constrained ECUs |
+| Secondary Full Verification ECU | Full Uptane verification for its images | in-toto verification for its images |
+| Resource Constrained ECU | Partial Uptane verification for its images | No in-toto verification |
+
+**Table:** Comparison of responsibilities expected of different ECUs where
+in-vehicle communication bandwidth is not constrained. As before, the in-toto
+verification process used can be either the full workflow or the stop-gap
+option, as this choice does not affect individual ECUs, but only the setup as a
+whole.
+
+Note that the stop-gap option can be employed in conjunction with the solutions
+for the other constraints. A vehicle may simultaneously have no ECUs capable of
+full in-toto verification _and_ limited bandwidth for in-vehicle communication.
+Similarly, a vehicle may have no ECUs capable of full in-toto verification _and_
+some of these ECUs may be incapable of any in-toto verification at all.
+
+## Security Analysis
+
+Scudo is an extension for Uptane that adds in-toto's software supply chain
+security properties. Therefore, its threat model is a combination of the threat
+models of Uptane for the delivery of the software artifacts, and in-toto for the
+development of the image as it passes through the software supply chain.
+
+## Backwards Compatibility
+
+Implementers of Scudo should always consider backwards compatibility. In itself,
+this PURE should not introduce any backwards incompatibility to existing Uptane
+deployments. However, introducing in-toto into the mix can complicate matters.
```

Comment:
How can it complicate matters?

Reply:
I updated the full section and noted some specific points of concern.